### PR TITLE
resolve local env vars

### PR DESCRIPTION
**Pull request by mahboubii**

### CODE REVIEW

File: browser-test-helper.js [hunk 1]
```diff
@@ -1,7 +1,8 @@
...
 
-window.pkg = pkg;
-window.STREAM_API_KEY = process.env.STREAM_API_KEY;
-window.STREAM_ANALYTICS_TOKEN = process.env.STREAM_ANALYTICS_TOKEN;
-window.STREAM_BASE_URL = process.env.STREAM_BASE_URL;
+window.env = {
+    STREAM_API_KEY: process.env.STREAM_API_KEY,
+    STREAM_ANALYTICS_TOKEN: process.env.STREAM_ANALYTICS_TOKEN,
+    STREAM_BASE_URL: process.env.STREAM_BASE_URL,
+};
```
Comment (ferhatelmas) [inline browser-test-helper.js:8]: :+1:

Comment (ferhatelmas) [reply browser-test-helper.js:8]: but maybe with `stream_` prefix, window is global at the end. We don't want to mess others' env.

Comment (ferhatelmas) [reply browser-test-helper.js:8]: and then we don't need `STREAM_` per property.

Comment (mahboubii) [reply browser-test-helper.js:8]: fixed, could you update the github env vars?

Comment (ferhatelmas) [reply browser-test-helper.js:8]: himm, I think we should keep `STREAM_` prefix in environment variables but good to remove inside set env. Something like:

```js
window.__stream_env__ = {
  API_KEY: process.env.STREAM_API_KEY,
  ...
}
```

File: .gitignore [hunk 2]
```diff
@@ -12,4 +12,5 @@ node_modules
 .DS_Store
 .AppleDouble
 .LSOverride
-/lib
\ No newline at end of file
+/lib
+.env
```
Comment (ferhatelmas) [inline .gitignore:16]: how about end of line ?

File: karma.config.js [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
 
 // Webpack config tweaks for test-time
 delete webpackConfig.entry;
 delete webpackConfig.output;
 webpackConfig.devtool = 'inline-source-map';
-webpackConfig.plugins = [
-    new webpack.EnvironmentPlugin(['STREAM_API_KEY', 'STREAM_ANALYTICS_TOKEN', 'STREAM_BASE_URL']),
-];
+
+webpackConfig.plugins = [new webpack.EnvironmentPlugin(process.env)];
```
Comment (ferhatelmas) [inline karma.config.js:10]: shorter, better :+1:

**Review by ferhatelmas — APPROVED** [review]
LGTM

File: .gitignore (hunk 2 above)
Comment (mahboubii) [reply .gitignore:16]: IDE fails to add that for the dotfiles 👎

Comment (ferhatelmas) [reply .gitignore:16]: :dancing_men:

**Pull request merged**